recipes/repository.rb

```diff
@@ -165,12 +203,14 @@
   end
 
   # Add YUM repository
-  zypper_repository 'datadog' do
-    description 'datadog'
-    baseurl baseurl
-    gpgkey agent_major_version < 7 ? node['datadog']['yumrepo_gpgkey'] : node['datadog']['yumrepo_gpgkey_new']
-    gpgautoimportkeys false
-    gpgcheck false
-    action :create
+  rpm_gpg_keys.each do |rpm_gpg_key|
+    zypper_repository 'datadog' do
+      description 'datadog'
+      baseurl baseurl
+      gpgkey agent_major_version < 7 ? node['datadog']['yumrepo_gpgkey'] : node['datadog']["yumrepo_gpgkey_new_#{rpm_gpg_key[rpm_gpg_keys_short_fingerprint]}"]
+      gpgautoimportkeys false
+      gpgcheck false
+      action :create
+    end
```

Comment:
This will create one `zypper_repository` resource for each listed gpg key. Is that really what we want? 

Moreover, it seems to me that each call to the `zypper_repository` resource will overwrite the previous one (because they all have the same name), so only the last one would be taken into account.

Given that the keys are imported above anyway, you could do with only one `zypper_repository` resource, using one of the keys in `rpm_gpg_keys`.

Reply:
Just pushed a new commit with the fix. Thanks.